app/controllers/admin/articles_controller.rb

```diff
@@ -29,12 +29,20 @@ def update
       redirect_to admin_articles_path, notice: 'Premium article approved for publication'  
     elsif @article.rejected?
       redirect_to admin_articles_path, notice: 'Article not approved for publication'
-    elsif @article.for_revision?
+    elsif @article.for_revision? && URI(request.referer).path == admin_articles_path
```

Comment:
``` path = (request.referer || 'admin_article') ```
this looks like it is the new syntax :https://stackoverflow.com/questions/49912325/how-to-check-if-there-is-not-request-referer-in-rails

Reply:
Hmm - I think that code is doing something different, storing the request  referrer (as a string) or the path admin_article in the variable 'path'. The issue is making the request.referrer method return a path rather than an address...

Reply:
What exactly are you trying to do here @rupertlion I might be able to help if I understand.

Reply:
I'm trying to ensure that two different actions occur based on the submission of 2 different forms:

1) If article is reviewed (by editor), and is marked as for_revision, a comment is stored in the comment attribute of the article.

2) If article is edited (by journalist), due to the fact it is marked for_revision, the re-submission triggers the changing of the article status back to :pending and therefore adds it back to the approval (editor) workflow.

Of course we could use the user model in the future to direct these actions (editor vs journalist) but this isn't in place yet. Also, I think it is the fact the article is either reviewed or edited which is the salient action here.

Reply:
Yeah, don't overdo it at this stage. Wait until you have rale bases authorization implemented. It should be in your pipeline by now...